MetasysServices/MetasysClient.cs

```diff
@@ -1207,7 +1209,58 @@ private async Task SendCommandRequestAsync(Guid id, string command, IEnumerable<
             }
         }
 
+        #region "ad-hoc calls" // =========================================================================================================
+        // Support Ad-Hoc calls -----------------------------------------------------------------------------------------------------------
+        ///<inheritdoc/>
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage requestMessage, HttpCompletionOption completionOption = HttpCompletionOption.ResponseContentRead, CancellationToken cancellationToken = default)
+        {
+            var response = new HttpResponseMessage();
+            try
+            {
+                var baseUri = new Uri(Client.BaseUrl);
+
+                if (!baseUri.IsBaseOf(requestMessage.RequestUri))
+                {
+                    //By default it failed with 401 unauthorized. Do we need to handle it explicitly
+                    //Requesting to connect different host.
+                }
+
+                var flurlRequest = Client.Request();
+                flurlRequest.Url = GetUrlFromFromRequest(baseUri, requestMessage);
+
+                //Copy Headers
+                foreach (var header in requestMessage.Headers)
+                {
+                    flurlRequest.Headers.Add(header.Key, header.Value);

```

Comment:
Why do you use `Headers.Add`? I believe this could add multiple values for the same header name, which I don't think that's what you want.

Instead, I think you want to be able to add or replace any headers?

Flurl also has `WithHeader` and `WithHeaders`, which should also replace default headers on the client.